presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -389,6 +433,37 @@ else if (actualReturnType.isPrimitive()) {
 
             this.specializedTypeParameters = getDeclaredSpecializedTypeParameters(method);
 
+            allowedTypeParameters = ImmutableMap.<String, SubTypeSignature>builder()
```

Comment:
This seems rather hacky and brittle. Can we do this automatically? Or if this is information not available, we should move this to be part of types.